src/Microsoft.DotNet.Darc/src/Darc/Operations/GetHealthOperation.cs

```diff
@@ -239,11 +237,60 @@ public override async Task<int> ExecuteAsync()
         }
 
         /// <summary>
-        /// 
+        ///     Compute product dependency cycle metrics based on the input repositories and channels.
         /// </summary>
-        /// <param name="channels"></param>
+        /// <param name="channelsToEvaluate"></param>
         /// <returns></returns>
-        private HashSet<string> ComputeChannelsToEvaluate(IEnumerable<Channel> channels)
+        private List<Func<Task<HealthMetricWithOutput>>> ComputeProductDependencyCycleMetricsToRun(HashSet<string> channelsToEvaluate,
+            HashSet<string> reposToEvaluate, IEnumerable<Subscription> subscriptions, IEnumerable<DefaultChannel> defaultChannels)
+        {
+            IRemoteFactory remoteFactory = new RemoteFactory(_options);
+
+            // Compute the combinations that make sense.
+            var defaultChannelsRepoBranchCombinations = defaultChannels
+                .Where(df => channelsToEvaluate.Contains(df.Channel.Name))
+                // Replace refs/heads/
+                .Where(df => reposToEvaluate.Contains(df.Repository))
+                .Select<DefaultChannel, (string repo, string branch)>(df => (df.Repository, df.Branch.Replace("refs/heads/", "")));
+
+            var repoBranchCombinations = subscriptions
+                .Where(s => reposToEvaluate.Contains(s.TargetRepository))
+                .Where(s => channelsToEvaluate.Contains(s.Channel.Name))
+                .Select<Subscription, (string repo, string branch)>(s => (s.TargetRepository, s.TargetBranch))
+                .ToHashSet();
+
+            repoBranchCombinations.UnionWith(defaultChannelsRepoBranchCombinations);
```

Comment:
This is now duplicated in the other health metric. Seems like something that can be extracted.